src/execution_plan/ops/op_node_by_label_scan.c

```diff
@@ -48,28 +48,30 @@ void NodeByLabelScanOp_SetIDRange(NodeByLabelScan *op, UnsignedRange *id_range)
 	op->op.name = "Node By Label and ID Scan";
 }
 
-static void _ConstructIterator(NodeByLabelScan *op) {
+static GrB_Info _ConstructIterator(NodeByLabelScan *op, Schema *schema) {
 	GraphContext *gc = QueryCtx_GetGraphCtx();
-	Schema *schema = GraphContext_GetSchema(gc, op->n->label, SCHEMA_NODE);
-	assert(schema);
 	GxB_MatrixTupleIter_new(&op->iter, Graph_GetLabelMatrix(gc->g, schema->id));
 	NodeID minId = op->id_range->include_min ? op->id_range->min : op->id_range->min + 1;
 	NodeID maxId = op->id_range->include_max ? op->id_range->max : op->id_range->max - 1 ;
-	GxB_MatrixTupleIter_iterate_range(op->iter, minId, maxId);
-
+	return GxB_MatrixTupleIter_iterate_range(op->iter, minId, maxId);
 }
 
 static OpResult NodeByLabelScanInit(OpBase *opBase) {
```

Comment:
Please re-structure,
Start by setting default and handle each case individually.